provider/azure/environ.go

```diff
@@ -1074,7 +1074,89 @@ func (env *azureEnviron) instances(
 
 // AdoptResources is part of the Environ interface.
 func (env *azureEnviron) AdoptResources(controllerUUID string, fromVersion version.Number) error {
-	return errors.NotImplementedf("AdoptResources")
+	groupClient := resources.GroupsClient{env.resources}
+	resourceClient := resources.Client{env.resources}
+	var failed []string
+
+	apiVersions, err := collectAPIVersions(env.resources)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	resources, err := groupClient.ListResources(env.resourceGroup, "", nil)
```

Comment:
I know it's ugly and gross, but the API calls should be wrapped with env.callAPI to take care of rate limiting.